client/app/bundles/course/assessment/question/programming/containers/OnlineEditor/Java/OnlineEditorJavaView.jsx

```diff
@@ -0,0 +1,489 @@
+              expectEquals: <code>{intl.formatMessage(javaTranslations.expectEquals)}</code>,
+            }}
+          />
+        </div>
+        <div style={{ marginBottom: '0.5em' }}>
+          <FormattedMessage
+            id="course.assessment.question.programming.onlineEditorJavaView.testCasesCodeDescription"
+            defaultMessage={
+              '{editor}: Clicking {code} will toggle a code editor for you to write code into each ' +
+              'test case. This allows you to initialize variables and all functions for each test. ' +
+              'For example: {codeExample}' +
+              '{codeExampleExpression} and {codeExampleExpected} can then be input into {expression} and ' +
+              '{expected} respectively.'
+            }
+            values={{
+              expression: <b>{intl.formatMessage(translations.expressionHeader)}</b>,
+              expected: <b>{intl.formatMessage(translations.expectedHeader)}</b>,
+              code: <code>{intl.formatMessage(javaTranslations.testCaseDescriptionCode)}</code>,
+              editor: <code>{intl.formatMessage(javaTranslations.testCaseDescriptionEditor)}</code>,
+              /* eslint-disable react/jsx-indent */
+              codeExample:
+                <pre style={{ marginTop: '0.5em', paddingTop: '3px', paddingBottom: '3px' }}>
+                  <p style={{ marginBottom: 0 }}>
+                    {'int arr [] = {0,0,0}; // Initialize variables'}
+                  </p>
+                  <p style={{ marginBottom: 0 }}>{'addOneToArray(arr); // Make function calls'}</p>
+                  <p style={{ marginBottom: 0 }}>{'int expected [] = {1,1,1}; // Make function calls'}</p>
+                  <p style={{ marginBottom: 0 }}>
+                    {'setAttribute("expression", "addOneToArray([0,0,0])");'}
+                    {' // Override the default expression displayed'}
+                  </p>
+                </pre>,
+              /* eslint-enable react/jsx-indent */
+              codeExampleExpected: <code>{'expected'}</code>,
+              codeExampleExpression: <code>{'arr'}</code>,
+            }}
+          />
+        </div>
+        { errorTextElement }
+        {
+          this.renderTestCases(intl.formatMessage(translations.publicTestCases),
+          testCases, 'public')
+        }
+        {
+          this.renderTestCases(intl.formatMessage(translations.privateTestCases),
+          testCases, 'private')
+        }
+        {
+          this.renderTestCases(intl.formatMessage(translations.evaluationTestCases),
+          testCases, 'evaluation')
+        }
+      </div>
+    );
+  }
+
+  render() {
+    const { intl, autograded, isLoading, hasSubmissions } = this.props;
+    const submitAsFile = this.props.data.get('submit_as_file');
+    let toggleLabel = intl.formatMessage(javaTranslations.submitAsFileToggle);
+    if (hasSubmissions) {
+      toggleLabel = intl.formatMessage(javaTranslations.submitAsFileToggleDisabled);
```

Comment:
Yes, I understand that the backend is different for these two modes.  That's why "Submit as file" is confusing, it doesn't tell the user that the mode is fundamentally different and the way student code is evaluated is also different. User might think that it's only about edit online and upload file from offline.

Perhaps name it "Single file/class mode", And add some explaintion to both modes ?@weiqingtoh might have a better naming suggestion :-)